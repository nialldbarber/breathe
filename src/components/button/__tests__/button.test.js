import React from 'react';
import { mount, shallow } from 'enzyme';
import Button from '~/components/button';

describe('<Button />', () => {
  const buttonText = 'Click me!';
  let component;
  beforeEach(() => {
    component = mount(<Button type="button" text={buttonText} fn={() => {}} />);
  });

  test('To match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('Renders text provided', () => {
    const text = component.find('button').text();
    expect(text).toBe(buttonText);
  });

  test('Event fires when clicked', () => {
    const mockCallback = jest.fn();

    const buttonClicked = shallow(<Button fn={mockCallback} />);
    buttonClicked.find('button').simulate('click');
    expect(mockCallback.mock.calls.length).toEqual(1);
  });
});
