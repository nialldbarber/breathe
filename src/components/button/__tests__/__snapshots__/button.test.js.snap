// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> To match snapshot 1`] = `ReactWrapper {}`;
